pkg/config/store.go

```diff
@@ -78,6 +78,10 @@ type ChangeNotifier interface {
 	// The client should issue ReadChangeLog to see what has changed if the call is available.
 	// else it should re-read the store, perform diff and apply changes.
 	RegisterStoreChangeListener(s StoreListener)
+
+	// IsStoreChangeAvailable returns true only when the backend store is configured
+	// to emit change notifications.
+	IsStoreChangeAvailable() bool
```

Comment:
What I'm thinking is same as yours. But the point is, how `manager` reacts upon `NotifyStoreChange`. I uploaded two WIP PRs to clarify the approaches:
1. https://github.com/jmuk/mixer/commit/f9c680836ef4d30a7be7e74344c1b63383fc9aa2
   - `NotifyStoreChange` directly invokes `fetchAndNotify`. otherwise `go c.loop()` will do the polling.
2. https://github.com/jmuk/mixer/commit/f574f180dbf818033ccf66e0c4b01f72b6b0ec19
   - `NotifyStoreChange` just remembers the index, the polling loop invokes `fetchAndNotify` (but if it knows no changes, it can skip `fetchAndNotify` invocation)

I was thinking the former approach originally, and this needs `IsStoreChangeAvailable` method so that manager can fall back to `loop` method when the change events are not enabled on the current storage.